src/app/shared/users-list/users-list.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { UsersService } from 'src/app/shared/users.service';
+import { User } from '../users.model';
+
+@Component({
+  selector: 'app-users-list',
+  templateUrl: './users-list.component.html',
+  styleUrls: ['./users-list.component.css']
+})
+export class UsersListComponent implements OnInit {
+
+  usersList: User[];
+  @Input() route: String;
+
+  constructor(private usersService: UsersService) { }
+
+  ngOnInit(): void {
+
+    if (this.route === 'active-users') {
+      this.usersList = this.usersService.getActiveUsers();
+    }
+    else if (this.route === 'deleted-users') {
+      this.usersList = this.usersService.getDeletedUsers();
+    }
+    else if (this.route === 'manage-users') {
+      this.usersList = this.usersService.getAllUsers();
+    }
```

Comment:
This is a dummy component. It should only receive data and output it. It should only have on input `users` which is an array of type `User`. The `active`, `deleted` and `manage` components should take the responsibility of sending the data to this component

Reply:
Removed all these and getting the data from their respective parent